.github/actions/npm_install/action.yml

```diff
@@ -0,0 +1,47 @@
+name: 'NPM Install with Retries'
+description: 'Run npm install with exponential backoff retries'
+
+inputs:
+    working-directory:
+        description: 'Directory where to run npm install'
+        required: false
+        default: '.'
+    max-attempts:
+        description: 'Maximum number of retry attempts'
+        required: false
+        default: '5'
+    initial-delay:
+        description: 'Initial delay in seconds before first retry'
+        required: false
+        default: '1'
+    max-delay:
+        description: 'Maximum delay in seconds between retries'
+        required: false
+        default: '60'
+
+runs:
+    using: 'composite'
+    steps:
+        - shell: bash
+          run: |
+              attempt=1
+              max_attempts=${{ inputs.max-attempts }}
+              delay=${{ inputs.initial-delay }}
+              max_delay=${{ inputs.max-delay }}
+
+              until cd ${{ inputs.working-directory }} && npm install; do
```

Comment:
Manually test this by putting in a bad command with appropriate logs, make sure the exponential backoff works exactly as expected. Try making it succeed at various points as well

Reply:
done